File: apps/server/src/config/board.config.ts
import type { BoardCell, ProjectCell, CellType } from '@kien-quoc/shared'

type CellDef = Omit<BoardCell, 'placements'>

export const BOARD_CELLS: CellDef[] = [
  // Row 0
  { id: 'cell-0-0', position: { row: 0, col: 0 }, type: 'competitive', name: 'Cửa khẩu Lạng Sơn', description: 'Cửa khẩu quan trọng với Trung Quốc' },
  { id: 'cell-0-1', position: { row: 0, col: 1 }, type: 'synergy', name: 'Đại học Bách khoa', description: 'Trung tâm đào tạo kỹ thuật' },
  { id: 'cell-0-2', position: { row: 0, col: 2 }, type: 'synergy', name: 'Viện Nghiên cứu', description: 'Nghiên cứu khoa học công nghệ' },
  { id: 'cell-0-3', position: { row: 0, col: 3 }, type: 'competitive', name: 'Khu CN Việt Trì', description: 'Khu công nghiệp phía Bắc' },
  
  // Row 1 (cells 1,2 are project)
  { id: 'cell-1-0', position: { row: 1, col: 0 }, type: 'shared', name: 'Đồng bằng sông Hồng', description: 'Vùng nông nghiệp trù phú' },
  { id: 'cell-1-3', position: { row: 1, col: 3 }, type: 'shared', name: 'Cảng Hải Phòng', description: 'Cảng biển lớn miền Bắc' },
  
  // Row 2 (cells 1,2 are project)
  { id: 'cell-2-0', position: { row: 2, col: 0 }, type: 'shared', name: 'Cảng Đà Nẵng', description: 'Cảng biển miền Trung' },
  { id: 'cell-2-3', position: { row: 2, col: 3 }, type: 'shared', name: 'Tây Nguyên', description: 'Vùng cao nguyên trung bộ' },
  
  // Row 3
  { id: 'cell-3-0', position: { row: 3, col: 0 }, type: 'competitive', name: 'Khu CN Biên Hòa', description: 'Khu công nghiệp lớn miền Nam' },
  { id: 'cell-3-1', position: { row: 3, col: 1 }, type: 'cooperation', name: 'Đồng bằng sông Cửu Long', description: 'Vựa lúa của cả nước' },
  { id: 'cell-3-2', position: { row: 3, col: 2 }, type: 'cooperation', name: 'Trung tâm Tài chính', description: 'Trung tâm tài chính TP.HCM' },
  { id: 'cell-3-3', position: { row: 3, col: 3 }, type: 'competitive', name: 'Cảng Cát Lái', description: 'Cảng container lớn nhất' },
]

export function createProjectCell(name: string, description: string): ProjectCell {
  return {
    id: 'project-center',
    name,
    description,
    totalContributed: 0,
    contributingTeams: [],
  }
}

export function initBoardCells(): BoardCell[] {
  return BOARD_CELLS.map(cell => ({ ...cell, placements: [] }))
}

File: backend/config/board_config.py
"""Board configuration."""

from pydantic import BaseModel

from backend.config.scoring_config import CellType


class CellConfig(BaseModel):
    name: str
    type: CellType
    indices: list[str]


# 16 cells on 4x4 board
BOARD_CELLS: dict[tuple[int, int], CellConfig] = {
    (0, 0): CellConfig(
        name='Cửa khẩu Lạng Sơn',
        type=CellType.COOPERATION,
        indices=['integration', 'economy'],
    ),
    (0, 1): CellConfig(
        name='Đại học Bách khoa',
        type=CellType.SYNERGY,
        indices=['science', 'society'],
    ),
    (0, 2): CellConfig(
        name='Viện Hàn lâm',
        type=CellType.SYNERGY,
        indices=['science', 'culture'],
    ),
    (0, 3): CellConfig(
        name='Khu CN Việt Trì',
        type=CellType.COMPETITIVE,
        indices=['economy', 'environment'],
    ),
    (1, 0): CellConfig(
        name='Đồng bằng sông Hồng',
        type=CellType.SHARED,
        indices=['society', 'environment'],
    ),
    (1, 1): CellConfig(
        name='Dự án Quốc gia',
        type=CellType.PROJECT,
        indices=[],
    ),
    (1, 2): CellConfig(
        name='Dự án Quốc gia',
        type=CellType.PROJECT,
        indices=[],
    ),
    (1, 3): CellConfig(
        name='Cảng Đà Nẵng',
        type=CellType.COMPETITIVE,
        indices=['economy', 'integration'],
    ),
    (2, 0): CellConfig(
        name='Tây Nguyên',
        type=CellType.SYNERGY,
        indices=['environment', 'economy'],
    ),
    (2, 1): CellConfig(
        name='Dự án Quốc gia',
        type=CellType.PROJECT,
        indices=[],
    ),
    (2, 2): CellConfig(
        name='Dự án Quốc gia',
        type=CellType.PROJECT,
        indices=[],
    ),
    (2, 3): CellConfig(
        name='KCX Tân Thuận',
        type=CellType.COMPETITIVE,
        indices=['economy', 'science'],
    ),
    (3, 0): CellConfig(
        name='Đồng bằng Cửu Long',
        type=CellType.SHARED,
        indices=['society', 'economy'],
    ),
    (3, 1): CellConfig(
        name='Khu đô thị Thủ Đức',
        type=CellType.SYNERGY,
        indices=['society', 'science'],
    ),
    (3, 2): CellConfig(
        name='Trung tâm Tài chính',
        type=CellType.COOPERATION,
        indices=['economy', 'integration'],
    ),
    (3, 3): CellConfig(
        name='Cảng Sài Gòn',
        type=CellType.COMPETITIVE,
        indices=['economy', 'integration'],
    ),
}

# Project cells (4 center cells)
PROJECT_CELLS: list[tuple[int, int]] = [(1, 1), (1, 2), (2, 1), (2, 2)]
PROJECT_CELL_ID = 'project-center'


def get_regular_cells() -> dict[tuple[int, int], CellConfig]:
    """Get cells excluding project cells."""
    return {k: v for k, v in BOARD_CELLS.items() if k not in PROJECT_CELLS}


def create_cell_id(pos: tuple[int, int]) -> str:
    """Create cell ID from position."""
    return f'cell-{pos[0]}-{pos[1]}'
